3.6 release blog post (#148)
sourcegraph/about: blogposts/announcing-sourcegraph-3.6.md

  </div>
  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
</p>
-->

Admins can now install the [Sourcegraph for Bitbucket Server plugin](https://github.com/sourcegraph/bitbucket-server-plugin) to their Bitbucket Server instance to provide code intelligence automatically to their teams. Code intelligence, previously available only through the browser extension, includes hovers, tooltips, and line decorations while browsing and reviewing code in Bitbucket Server.

Previously, many users were not aware of the Sourcegraph browser extension, or had trouble configuring it to use their company’s Sourcegraph instance. Now, admins can provide an easier onboarding path for all team members who want to browse or review code on Bitbucket Server.

There are two configuration steps:

1. Install the [Sourcegraph for Bitbucket Server plugin](https://github.com/sourcegraph/bitbucket-server-plugin)
2. Update the [Bitbucket Server external service configuration](https://docs.sourcegraph.com/admin/external_service/bitbucket_server#native-extension) in your Sourcegraph instance to include `”corsOrigin”: “<bitbucket server url>”`

> ## 30% search latency improvement for small and medium instances

Reviewer: ```suggestion
## 30% indexed search latency improvement for small and medium instances
```

---
title: 'Sourcegraph 3.6: Bitbucket Server plugin, search performance improvements, and quick links'
author: Christina Forney
publishDate: 2019-07-20T10:00-07:00
tags: [
  blog
]
slug: sourcegraph-3.6
heroImage: /sourcegraph-mark.png
published: true
---

> [Sourcegraph](https://about.sourcegraph.com/) is the standard developer platform for code search and navigation at many of the largest and most exacting technology companies. With Sourcegraph, every company has access to the same kind of tools that Google and Facebook developers use every day.

Reviewer 2: ```suggestion
Sourcegraph is the standard developer platform for code search and navigation at many of the largest and most exacting technology companies. With Sourcegraph, every company has access to the same kind of tools that Google and Facebook developers use every day.
```

Author: This was intro in the previous releases, we don't want to include a link to the about site?

Reviewer 2: Yeah, I'm not sure why I put it in here. If you want to keep it, just change the url to `/`.

<div style="padding-left: 2rem">

> [**🌎 Sourcegraph plugin provides native code intelligence to Bitbucket Server**](#)<br />

Reviewer 2: ```suggestion
[**🌎 Bitbucket Server integration brings code intelligence to pull requests and more**](#)<br />
```

While it is a Bitbucket Server plugin, from Sourcegraph's point of view, it is an integration.

Author: I wanted to make it clear that it was Sourcegraph providing the code intelligence. Also, we had code intelligence before via the browser extension, and I want to distinguish the difference. Any thoughts on how to improve that? What about 'Native code intelligence with Sourcegraph Bitbucket Server plugin'

Reviewer 2: > What about 'Native code intelligence with Sourcegraph Bitbucket Server plugin

Yep, that's awesome!

[**📉 30% indexed search latency improvement for small and medium instances**](#)<br />

> [**🔗 Add quick links to repositories or important resources below the search bar**](#)<br />

Reviewer 2: ```suggestion
[**🔗 New customizable quick links for frequently used code searches or other URLs**](#)<br />
```

Suggested this change as I imagine most would use it for searches they use frequently, and I don't think you need to mention in this heading where the quick links go.

> [**☁️ Added support for Bitbucket Cloud external service configuration**](#)<br />

Reviewer 2: ```suggestion
[**☁️ New repository syncing support for Bitbucket Cloud**](#)<br />
```

> [**🛠 Improved external service configuration and syncing experience for admins**](#)<br />

Reviewer 2: ```suggestion
[**🛠 Improved repository syncing progress UI**](#)<br />
```

> [**🙅‍♀️ Easily exclude GitHub repositories from syncing (by regexp)**](#)<br />

Reviewer 2: ```suggestion
[**🙅‍♀️ Exclude rules for GitHub repository sync configuration now support reg exp.**](#)<br />
```

> [**✈️ Meet us in person at GopherCon!**](#)<br />

Reviewer 2: ```suggestion
[**✈️ Meet the team at GopherCon 2019!**](#)<br />
```

[**📝 Changelog**](#35-changelog)<br />

[**🎖️ Thank you**](#thank-you)<br />

</div>

**Deploy or upgrade:** [Local](https://docs.sourcegraph.com/#quickstart-guide) | [AWS](https://github.com/sourcegraph/deploy-sourcegraph-aws) | [DigitalOcean](https://marketplace.digitalocean.com/apps/sourcegraph?action=deploy&refcode=48dfb3ccb51c) | [Kubernetes cluster](https://github.com/sourcegraph/deploy-sourcegraph)

## Sourcegraph plugin provides native code intelligence to Bitbucket Server

TO DO: VIDEO

<!--
<p class="container">
  <div style="padding:56.25% 0 0 0;position:relative;">
    <iframe src="https://player.vimeo.com/video/{ID}?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
  </div>
  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
</p>
-->

> Admins can now install the [Sourcegraph for Bitbucket Server plugin](https://github.com/sourcegraph/bitbucket-server-plugin) on their Bitbucket Server instance to provide code intelligence automatically to their teams. Code intelligence, previously available only through the browser extension, includes hovers, tooltips, and line decorations while browsing and reviewing code in Bitbucket Server.

Reviewer 2: ```suggestion
The [Sourcegraph for Bitbucket Server plugin](https://github.com/sourcegraph/bitbucket-server-plugin) adds hovers tooltips, go to definition, and custom line decorations for all users when browsing and reviewing code on Bitbucket Server instances.
```

Author: What do you think of this instead? I want to make it clear that the plugin is the new thing, not the ability to have code intelligence.

```suggestion
The [Sourcegraph for Bitbucket Server plugin](https://github.com/sourcegraph/bitbucket-server-plugin) is now available for installation on Bitbucket Server instances to provide native code intelligence automatically. Code intelligence, previously available only through the browser extension, includes hovers, tooltips, go to definition, and custom line decorations when browsing and reviewing code on Bitbucket Server instances.
```

Reviewer 2: I love it!

> Previously, many users were not aware of the Sourcegraph browser extension. Now, admins can provide an easier onboarding path for all team members who want to browse or review code on Bitbucket Server.

Reviewer 2: ```suggestion
```

In this case, I don't think it's worth stating the problem as most uses didn't know they were missing anything.

There are two configuration steps:

1. Install the [Sourcegraph for Bitbucket Server plugin](https://github.com/sourcegraph/bitbucket-server-plugin)
2. Update the [Bitbucket Server external service configuration](https://docs.sourcegraph.com/admin/external_service/bitbucket_server#native-extension) in your Sourcegraph instance to include `”corsOrigin”: “<bitbucket server url>”`

## 30% indexed search latency improvement for small and medium instances

![chart showing 30% improvement of search latency](images/3.6-search-latency.png "30% improvement of search latency")

> We have made several changes to our search algorithm that improve performance and reduce peak memory usage. Our benchmarks show that the **99th percentile latency** of indexed searches in small to medium instances (less than 3000 repositories) **is reduced by up to 30%**. We will be continuing work to further benchmark and optimize Sourcegraph for our customers with 30,000+ repositories.

Reviewer 2: I can't think of a good alternative, but I don't like the "we made several changes" part because I instantly want to know **what** the changes were. This would be fine if we had a blog post that contained the changes and details but instead, can we link to an issue or PR which provides some level of detail?

Author: What about:

```suggestion
We have improved our search algorithm's performance and reduced peak memory usage. Our benchmarks show that the **99th percentile latency** of indexed searches in small to medium instances (less than 3000 repositories) **is reduced by up to 30%**. We will be continuing work to further benchmark and optimize Sourcegraph for our customers with 30,000+ repositories.
```

Reviewer 2: Brilliant!

## Add quick links to repositories or important resources below the search bar

TO DO: VIDEO

<!--
<p class="container">
  <div style="padding:56.25% 0 0 0;position:relative;">
    <iframe src="https://player.vimeo.com/video/{ID}?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
  </div>
  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
</p>
-->

> [Quick links](https://docs.sourcegraph.com/user/quick_links) can now be added below the search bar on the home page and search results page. This makes it easier for teams to quickly access the Sourcegraph pages they care most about such as your team’s main repository, or have easy access to external resources like your CI system or team documentation.

Reviewer 2: ```suggestion
[Quick links](https://docs.sourcegraph.com/user/quick_links) can now be added below the search bar on the home and search results pages. They can be used for quickly accessing frequently used URLs such as a code search, main repository, or external resources like your CI system or team documentation.
```

Not totally happy with the above, but I think adding an example of linking to a code search is important.

> Add a quick link to global, organization, or user settings with the `quicklinks` field:

Reviewer 2: ```suggestion
Quick links can be defined at the global, organization, and user level using the `quicklinks` field:
```

```
{
  // ...

>   "quicklinks": [

Reviewer 2: Add a code search example here

Author: Great idea!

    {
      "name": "Sourcegraph docs",
      "url": "https://docs.sourcegraph.com"
    },
    {
      "name": "Main repository",
      "url": "/github.com/sourcegraph/sourcegraph"
    }
  ]
  // ...
}
```

## Added support for Bitbucket Cloud external service configuration

TO DO: VIDEO

<!--
<p class="container">
  <div style="padding:56.25% 0 0 0;position:relative;">
    <iframe src="https://player.vimeo.com/video/{ID}?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
  </div>
  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
</p>
-->

> Previously, Bitbucket Cloud repositories had to be added and configured as individual git repositories. In Sourcegraph 3.6 we added first class support for connecting [Bitbucket Cloud repositories](https://docs.sourcegraph.com/admin/external_service/bitbucket_cloud). In this release, all repositories will be added for the username configured, and teams listed that the Bitbucket Cloud app password grants access to.

Reviewer 2: ```suggestion
Sourcegraph 3.6 now supports repository syncing for [Bitbucket Cloud](https://docs.sourcegraph.com/admin/external_service/bitbucket_cloud). In this initial release, all repositories accessible by the token for the user (including associated teams) will be synced. Support for excluding repositories is slated for 3.7.
```

@Unknwon, @christinaforney I'm presuming we are adding exclude rule support in a future release?

Author: Yes, but don't want to specify which release since 3.7 is such a short one for us with GopherCon + hack week.

## Improved external service configuration and syncing experience for admins

TO DO: Updated video (should we mark the 3.5 video as deprecated?)

<!--
<p class="container">
  <div style="padding:56.25% 0 0 0;position:relative;">
    <iframe src="https://player.vimeo.com/video/{ID}?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
  </div>
  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
</p>
-->

> Admins no longer have to wait for an external service to finish syncing before navigating away from the configuration page. Previously, this was problematic for customers adding external services with lots of repositories. Now, the syncing continues in the background and progress is shown to admins in the global navigation bar’s syncing status indicator.

Reviewer 2: ```suggestion
Applying an external service configuration change is now done in the background upon save, with progress information available to admins by clicking the syncing status indicator on the global navigation bar.
```

> The experimental repository syncing status indicator announced in 3.5 has been improved enough to turn this feature on for all site admins by default. Improvements include handling scale of 30,000+ repositories, and only showing the number of repositories remaining to be cloned.

Reviewer 2: I don't think
```suggestion
This progress status UI has been improved to support instances with 30,000+ repositories, only showing the number of repositories remaining to be cloned.
```

Author: Let me know about the change I made as an alternative.

## Easily exclude GitHub repositories from syncing (by regexp)

> The `github.exclude` setting in [GitHub external service config](https://docs.sourcegraph.com/admin/external_service/github#configuration) has been expanded to handle regular expressions using `{"pattern": "^topsecretrepogroup/.*"}`.

Reviewer 2: ```suggestion
The `github.exclude` field for the [GitHub external service](https://docs.sourcegraph.com/admin/external_service/github#configuration) now supports the use of regular expressions using `{"pattern": "^topsecretrepogroup/.*"}`.
```

> Previously, only names or IDs could be specified. Now, it is easier to exclude entire groups of repositories, such as repositories containing secrets, or data projects that aren’t relevant for code browsing.

Reviewer 2: ```suggestion
Now, entire groups of repositories such as those containing secrets, can be excluded in a single regexp instead of manually entering the name or id for each repository.
```

Quick links can be defined at the global, organization, and user level using the `quicklinks` field:

```
{
  // ...
  "quicklinks": [
    // An external doc resource
    {
      "name": "Sourcegraph docs",
      "url": "https://docs.sourcegraph.com"
    },
    // The main repository for your organization
    {
      "name": "Main repository",
      "url": "/github.com/sourcegraph/sourcegraph"
    },
    // A commonly used search (all CHANGELOG updates)
    {
      "name": "Changelog updates",
      "url": "/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:changelog.md+type:diff"
    }
  ]
  // ...
}
```

## New repository syncing support for Bitbucket Cloud

<p class="container">
  <div style="padding:56.25% 0 0 0;position:relative;">
    <iframe src="https://player.vimeo.com/video/349145191?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
  </div>
  <p style="text-align: center"><a href="https://vimeo.com/349145191" target="_blank">View on Vimeo</a></p>
</p>

> Sourcegraph 3.6 now supports repository syncing for [Bitbucket Cloud](https://docs.sourcegraph.com/admin/external_service/bitbucket_cloud). In this initial release, all repositories accessible by the token for the user (including from teams listed in the config) will be synced. Support for excluding repositories is coming in a future version.

Author: ```suggestion
Sourcegraph 3.6 now supports repository syncing for [Bitbucket Cloud](https://docs.sourcegraph.com/admin/external_service/bitbucket_cloud). In this initial release, all repositories accessible by the token for the user (including from teams listed in the config) will be synced. Support for excluding repositories and for adding public repositories is coming in a future version.
```

[**🔗 New customizable quick links for frequently used code searches or other URLs**](#new-customizable-quick-links-for-frequently-used-code-searches-or-other-urls)<br />

[**☁️ New repository syncing support for Bitbucket Cloud**](#new-repository-syncing-support-for-bitbucket-cloud)<br />

[**🛠 Improved repository syncing progress UI**](#improved-repository-syncing-progress-ui)<br />

[**🙅‍♀️ Exclude rules for GitHub repository sync configuration now support regexp**](#exclude-rules-for-github-repository-sync-configuration-now-support-regexp)<br />

[**✈️ Meet the team at GopherCon 2019!**](#meet-the-team-at-gophercon-2019)<br />

[**📝 Changelog**](#36-changelog)<br />

[**🎖️ Thank you**](#thank-you)<br />

</div>

**Deploy or upgrade:** [Local](https://docs.sourcegraph.com/#quickstart-guide) | [AWS](https://github.com/sourcegraph/deploy-sourcegraph-aws) | [DigitalOcean](https://marketplace.digitalocean.com/apps/sourcegraph?action=deploy&refcode=48dfb3ccb51c) | [Kubernetes cluster](https://github.com/sourcegraph/deploy-sourcegraph)

## Native code intelligence with Sourcegraph Bitbucket Server plugin

> TO DO: VIDEO

Author: Taking this out so blogpost can go live, even if video is not complete.

```suggestion
```

      "url": "/github.com/sourcegraph/sourcegraph"
    },
    // A commonly used search (all CHANGELOG updates)
    {
      "name": "Changelog updates",
      "url": "/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:changelog.md+type:diff"
    }
  ]
  // ...
}
```

## New repository syncing support for Bitbucket Cloud

TO DO: VIDEO

<!--
<p class="container">
  <div style="padding:56.25% 0 0 0;position:relative;">
    <iframe src="https://player.vimeo.com/video/{ID}?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
  </div>
  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
</p>
-->

Sourcegraph 3.6 now supports repository syncing for [Bitbucket Cloud](https://docs.sourcegraph.com/admin/external_service/bitbucket_cloud). In this initial release, all repositories accessible by the token for the user (including from teams listed in the config) will be synced. Support for excluding repositories is coming in a future version.

## Improved repository syncing progress UI

> TO DO: Updated video (should we mark the 3.5 video as deprecated?)

Reviewer 2: I'll replace the video so it works in both posts
